benchflow-experiment-manager/application/src/main/java/cloud/benchflow/experimentmanager/services/external/faban/FabanManagerService.java

```diff
@@ -200,54 +202,75 @@ public RunId submitTrialToFaban(String experimentID, String trialID,
+    new Thread(() -> {
 
-    try {
+      // B) wait/poll for trial to complete and store the trial result in the DB
+      // TODO - is this the status we want to use? No it is a subset, should also
+      // include metrics computation status
+      RunStatus status;
+      FabanStatusRequest fabanStatusRequest = null;
 
-      // wait 60s before polling (Faban needs time to setup)
       try {
-        Thread.sleep(60 * 1000);
-      } catch (InterruptedException e) {
-        e.printStackTrace();
-      }
-
-      status = fabanClient.status(runId);
-
-      while (status.getStatus().equals(StatusCode.QUEUED)
-          || status.getStatus().equals(StatusCode.RECEIVED)
-          || status.getStatus().equals(StatusCode.STARTED)
-          || status.getStatus().equals(StatusCode.UNKNOWN)) {
 
+        // wait 60s before polling (Faban needs time to setup)
         try {
-          Thread.sleep(30 * 1000);
+          Thread.sleep(60 * 1000);
         } catch (InterruptedException e) {
           e.printStackTrace();
         }
 
         status = fabanClient.status(runId);
 
-      }
+        while (status.getStatus().equals(StatusCode.QUEUED)
+            || status.getStatus().equals(StatusCode.RECEIVED)
+            || status.getStatus().equals(StatusCode.STARTED)
+            || status.getStatus().equals(StatusCode.UNKNOWN)) {
+
+          try {
+            Thread.sleep(30 * 1000);
+          } catch (InterruptedException e) {
+            e.printStackTrace();
+          }
 
-      RunInfo runInfo = fabanClient.runInfo(runId);
+          status = fabanClient.status(runId);
 
-      return new FabanStatus(trialID, status.getStatus(), runInfo.getResult());
+        }
+
+        RunInfo runInfo = fabanClient.runInfo(runId);
+
+        fabanStatusRequest =
+            new FabanStatusRequest(trialID, status.getStatus(), runInfo.getResult());
+
+      } catch (FabanClientIOException | IllegalRunStatusException | MalformedURIException
+          | FabanClientBadRequestException | IllegalRunInfoResultException
+          | RunIdNotFoundException e) {
+        // TODO - handle me
+        e.printStackTrace();
+
+      } finally {
+
+        // in case there was some error
+        if (fabanStatusRequest == null) {
+          //See https://github.com/benchflow/benchflow/pull/473/files#r128371872
+          fabanStatusRequest = new FabanStatusRequest(trialID, StatusCode.UNKNOWN, Result.UNKNOWN);
+        }
+
+        // send status to experiment manager
+        TrialResource trialResource = BenchFlowExperimentManagerApplication.getTrialResource();
+
+        TrialIDElements trialIDElements = new TrialIDElements(trialID);
+
+        trialResource.setFabanResult(trialIDElements.getUsername(), trialIDElements.getTestName(),
+            trialIDElements.getTestNumber(), trialIDElements.getExperimentNumber(),
+            trialIDElements.getTrialNumber(), fabanStatusRequest);
+      }
+
+    }).start();
```

Comment:
This code is executed as part of the `DETERMINE_EXECUTE_TRIALS`, and since it is in a thread it will be asynchronous, which will make the state of the experiment change to `HANDLE_TRIAL_RESULT` and the while loop will finish. Once the FabanManager has a result it will then invoke the Scheduler again and then the life cycle continues.